### PR TITLE
Adding missing docs for existing routes for QnA Maker APIM

### PR DESCRIPTION
**Pull request by rokulka**
Adding missing docs for existing routes for QnA Maker APIM
  - Add /endpointsetttings path for service
Adding new docs for existing routes for QnA Maker Runtime
  - New swagger file as the "x-ms-parameterized-host" is different for them
  - Add /generateAnswer path for Runtime
  - Add /train path for Runtime

### CODE REVIEW

**Comment by tjprescott** [comment]
Numerous linter checks failing

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMakerRuntime.json [hunk 1]
```diff
@@ -0,0 +1,528 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "4.0",
+    "title": "QnAMaker Client",
+    "description": "An API for QnAMaker Service"
```
Comment (bingisbestest) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMakerRuntime.json]: runtime

File: specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMakerRuntime.json [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+          "Knowledgebases"
+        ],
+        "summary": "GenerateAnswer call to query the knowledgebase.",
+        "operationId": "Knowledgebase_GenerateAnswer",
+        "parameters": [
+          {
+            "$ref": "#/parameters/KbId"
+          },
+          {
+            "$ref": "#/definitions/QueryDTO"
```
Comment (bingisbestest) [inline specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMakerRuntime.json]: Needs to be a parameter def, see qnamaker.json

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by tjprescott — APPROVED** [review]
LGTM. However, please address @bingisbestest's comments.

**Review by ghost — APPROVED** [review]

**Pull request merged**